src/sys/socket/mod.rs

```diff
@@ -554,6 +556,11 @@ pub enum ControlMessageOwned {
     /// # }
     /// ```
     ScmTimestamp(TimeVal),
+    /// Nanoseconds resolution timestamp
+    ///
+    /// [Further reading](https://www.kernel.org/doc/html/latest/networking/timestamping.html)
+    #[cfg(all(target_os = "linux", any(target_arch = "x86_64"), target_env = "gnu"))]
```

Comment:
You should not restrict the target_arch to just x86_64.  libc defines SCM_TIMESTAMPNS for many architectures.

Reply:
This is the CI result when the conditional compilation was turned off: https://cirrus-ci.com/build/5978723427024896

I think there are cases where the symbols are added, but it was not supported: https://cirrus-ci.com/task/5721153533640704?command=main#L510

Reply:
It may be that Linux supports that socket option, but QEMU does not.  Looking at the Linux source code, I think it should be supported everywhere.

Reply:
I can add more arch if you have concrete ones in mind. Otherwise I think it makes sense to merge this as MVP since x86_64 covers major use cases, and follow up with other archs.

Reply:
You should add every arch that is supported by libc.  Don't be deterred by the fact that you can't run it in CI; some architectures don't have any CI coverage at all.  You have a working test case and there is no reason to believe that the feature would work differently on other architectures.

Reply:
Adding those would break CI, which prevents us from getting breaking warnings for future changes for those archs. Are you sure that is what we want?

Reply:
I'm not asking you to break CI.  I'm asking you to get the feature working on all platforms.  But if QEMU doesn't support it, then disable the test (not the feature) on those platforms.  There are already plenty of examples of tests that are disabled on QEMU platforms.